### PR TITLE
Add Software Licensing gists to the library

### LINKED ISSUE #146: Move SL gists to the Library
As part of some changes to the Software Licensing samples (showing how to support auto updates), I'm updating the [documentation we have on that](https://docs.easydigitaldownloads.com/article/383-automatic-upgrades-for-wordpress-plugins). There are quite a few gists in there, but I'd like to update them and move them to the Library, partly so I can edit them, and partly to simplify that documentation page a bit.

### PR DESCRIPTION
**Pull request by robincornett**
Fixes #146 

Proposed Changes:
1. Creates a new layout for sample code--currently just does not include the "To implement this snippet..." sentence which is at the end of the snippet layout.
2. Copies the five gists used in the HS docs for the SL Updater implementation.

Because the gists have been updated for SL 3.6.12 (specifically the updater instantiation one), I'm marking this as a draft for now.

### CODE REVIEW

**Comment by robincornett** [comment]
@ashleyfae I got Jekyll running locally, but it's not building the new pages, so I still have this set to draft. I'm not sure if it's just something I'm not doing correctly locally or if I've set things up incorrectly.

**Comment by ashleyfae** [comment]
@robincornett I've never used Jekyll before so it might be worth finding someone who knows how this works already so that we're not both fumbling around guessing. :D 

Did I see at one point that Sean was familiar with this?

**Comment by robincornett** [comment]
@ashleyfae never mind; I figured out what I was doing wrong. Had missed the opening line of each new file and the titles cannot include `:`. This is now working properly locally:

image

File: _extensions/software-licensing/updater-implementation-instantiate.html [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+---
+
+<?php
+
+/*
+Plugin Name: Sample Plugin
+Plugin URI: http://pippinsplugins.com/
+Description: Illustrates how to include an updater in your plugin for EDD Software Licensing
+Author: Pippin Williamson
+Author URI: http://pippinsplugins.com
```
Comment (ashleyfae) [inline _extensions/software-licensing/updater-implementation-instantiate.html:16]: Should we change these URL/author references to `https://sandhillsdev.com` and `Sandhills Development, LLC` ?

File: _extensions/software-licensing/updater-implementation-instantiate.html [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+License: GNU General Public License v2.0 or later
+License URI: http://www.gnu.org/licenses/gpl-2.0.html
+*/
+
+/**
+ * For further details please visit https://docs.easydigitaldownloads.com/article/383-automatic-upgrades-for-wordpress-plugins
+ */
+
+// this is the URL our updater / license checker pings. This should be the URL of the site with EDD installed
+define( 'EDD_SAMPLE_STORE_URL', 'http://easydigitaldownloads.com' ); // you should use your own CONSTANT name, and be sure to replace it throughout this file
```
Comment (ashleyfae) [inline _extensions/software-licensing/updater-implementation-instantiate.html:27]:
```suggestion
define( 'EDD_SAMPLE_STORE_URL', 'https://easydigitaldownloads.com' ); // you should use your own CONSTANT name, and be sure to replace it throughout this file
```

Just to set a good example. :D

File: _extensions/software-licensing/updater-implementation-settings.html [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+					<?php if( false !== $license ) { ?>
+						<tr valign="top">
+							<th scope="row" valign="top">
+								<?php _e('Activate License'); ?>
+							</th>
+							<td>
+								<?php if( $status !== false && $status == 'valid' ) { ?>
+									<span style="color:green;"><?php _e('active'); ?></span>
+									<?php wp_nonce_field( 'edd_sample_nonce', 'edd_sample_nonce' ); ?>
+									<input type="submit" class="button-secondary" name="edd_license_deactivate" value="<?php _e('Deactivate License'); ?>"/>
```
Comment (ashleyfae) [inline _extensions/software-licensing/updater-implementation-settings.html:46]:
```suggestion
									<input type="submit" class="button-secondary" name="edd_license_deactivate" value="<?php esc_attr_e('Deactivate License'); ?>"/>
```

File: _extensions/software-licensing/updater-implementation-settings.html [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+								<?php } else {
+									wp_nonce_field( 'edd_sample_nonce', 'edd_sample_nonce' ); ?>
+									<input type="submit" class="button-secondary" name="edd_license_activate" value="<?php _e('Activate License'); ?>"/>
```
Comment (ashleyfae) [inline _extensions/software-licensing/updater-implementation-settings.html:49]:
```suggestion
									<input type="submit" class="button-secondary" name="edd_license_activate" value="<?php esc_attr_e('Activate License'); ?>"/>
```

**Comment by ashleyfae** [comment]
Awesome @robincornett ! :raised_hands: Just left a few small notes.

**Comment by ashleyfae** [comment]
As per Slack discussion, we're going to hold off on the above changes for now, in light of them just being copy pasted from the existing docs at this point. We'll re-visit them later.

**Pull request merged**